cmd/kubeadm/app/cmd/join.go

```diff
@@ -203,7 +206,31 @@ func (j *Join) Run(out io.Writer) error {
 		return err
 	}
 
+	client, err := kubeconfigutil.KubeConfigToClientSet(cfg)
+	if err != nil {
+		return err
+	}
+	if err := kubeadmnode.ValidateAPIServer(client); err != nil {
```

Comment:
nit: this call could be inside the 1.7 conditional block since it doesn't need to happen if we're offloading bootstrapping to the 1.8+ kubelet.

Reply:
That will get removed in v1.9 anyway; it does no harm to print the server version unconditionally now ;)